common/src/main/java/com/wynntils/screens/WynntilsDialogueHistoryScreen.java

```diff
@@ -0,0 +1,297 @@
+        renderBackgroundTexture(poseStack);
+
+        // Make 0, 0 the top left corner of the rendered quest book background
+        poseStack.pushPose();
+        final float translationX = getTranslationX();
+        final float translationY = getTranslationY();
+        poseStack.translate(translationX, translationY, 1f);
+
+        renderTitle(poseStack, I18n.get("screens.wynntils.wynntilsDialogueHistory.title"));
+
+        renderVersion(poseStack);
+
+        renderButtons(poseStack, mouseX, mouseY, partialTick);
+
+        if (dialogues.isEmpty()) {
+            renderNoDialoguesHelper(poseStack);
+        } else {
+            renderCurrentPage(poseStack);
+        }
+
+        renderDescription(poseStack, I18n.get("screens.wynntils.wynntilsDialogueHistory.description"));
+
+        renderPageInfo(poseStack, getCurrentPage() + 1, getMaxPage() + 1);
+
+        poseStack.popPose();
+
+        renderTooltip(poseStack, mouseX, mouseY);
+    }
+
+    private void renderCurrentPage(PoseStack poseStack) {
+        List<TextRenderTask> textRenderTaskList = new ArrayList<>();
+        float maxWidth = Texture.QUEST_BOOK_BACKGROUND.width() / 2f - 20;
+
+        for (String line : dialogues.get(currentPage)) {
+            textRenderTaskList.add(new TextRenderTask(
+                    line,
+                    new TextRenderSetting(
+                            maxWidth,
+                            CommonColors.BLACK,
+                            HorizontalAlignment.Left,
+                            VerticalAlignment.Top,
+                            FontRenderer.TextShadow.NORMAL)));
+        }
+
+        FontRenderer.getInstance()
+                .renderTextsWithAlignment(
+                        poseStack,
+                        Texture.QUEST_BOOK_BACKGROUND.width() / 2f + 5,
+                        30,
+                        textRenderTaskList,
+                        maxWidth,
+                        Texture.QUEST_BOOK_BACKGROUND.height() - 50,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Middle);
+    }
+
+    @Override
+    public boolean mouseScrolled(double mouseX, double mouseY, double delta) {
+        setCurrentPage(getCurrentPage() + (delta > 0 ? -1 : 1));
+
+        return true;
+    }
+
+    @SubscribeEvent
+    public void onQuestsReloaded(QuestBookReloadedEvent.DialogueHistoryReloaded event) {
+        this.setDialogues(QuestBookManager.getDialogueHistory());
+    }
+
+    private void renderTooltip(PoseStack poseStack, int mouseX, int mouseY) {
+        if (this.hovered instanceof ReloadButton) {
+            RenderUtils.drawTooltipAt(
+                    poseStack,
+                    mouseX,
+                    mouseY,
+                    100,
+                    RELOAD_TOOLTIP,
+                    FontRenderer.getInstance().getFont(),
+                    true);
+            return;
+        }
+
+        if (this.hovered instanceof QuestsPageButton) {
```

Comment:
It seems like Qodana is correct. `hovered` is private, initialized to null, and I can't see any setters.

Reply:
Line 232?